mm/2s2h/Rando/ActorBehavior/EnemyDrops.cpp

```diff
@@ -0,0 +1,165 @@
+    { ACTOR_EN_BBFALL, RC_ENEMY_DROP_RED_BUBBLE },
+    { ACTOR_EN_RD, RC_ENEMY_DROP_REDEAD_GIBDO },
+    { ACTOR_EN_SB, RC_ENEMY_DROP_SHELLBLADE },
+    { ACTOR_EN_PR2, RC_ENEMY_DROP_SKULLFISH },
+    { ACTOR_EN_ST, RC_ENEMY_DROP_SKULLTULA },
+    { ACTOR_EN_SW, RC_ENEMY_DROP_SKULLWALLTULA },
+    { ACTOR_EN_KAME, RC_ENEMY_DROP_SNAPPER },
+    { ACTOR_EN_SKB, RC_ENEMY_DROP_STALCHILD },
+    { ACTOR_EN_TITE, RC_ENEMY_DROP_TEKTITE },
+    { ACTOR_EN_WF, RC_ENEMY_DROP_WOLFOS },
+    { ACTOR_EN_WALLMAS, RC_ENEMY_DROP_WALLMASTER },
+};
+
+RandoCheckId GetRandoCheckByActorId(int16_t actorId) {
+    for (auto& map : actorIdToRandoCheckIdMap) {
+        if (map.first == actorId) {
+            return map.second;
+        }
+    }
+
+    return RC_UNKNOWN;
+}
+
+Actor* FindActor(Vec3f position, ActorType actorType) {
+    ActorListEntry actorList = gPlayState->actorCtx.actorLists[actorType];
+
+    Actor* currentActor = actorList.first;
+    for (size_t i = 0; i < actorList.length; i++) {
+        if (currentActor->world.pos.x != position.x && currentActor->world.pos.z != position.z) {
+            currentActor = currentActor->next;
+        } else {
+            return currentActor;
+        }
+    }
+    return nullptr;
+}
+
+void SpawnEnemyDrop(Vec3f position, Actor* actor, RandoCheckId randoCheckId) {
+    CustomItem::Spawn(
+        position.x, position.y, position.z, 0,
+        CustomItem::KILL_ON_TOUCH | CustomItem::TOSS_ON_SPAWN | CustomItem::ABLE_TO_ZORA_RANG, randoCheckId,
+        [](Actor* actor, PlayState* play) {
+            RandoSaveCheck& randoSaveCheck = RANDO_SAVE_CHECKS[CUSTOM_ITEM_PARAM];
+            randoSaveCheck.eligible = true;
+        },
+        [](Actor* actor, PlayState* play) {
+            auto& randoSaveCheck = RANDO_SAVE_CHECKS[CUSTOM_ITEM_PARAM];
+            RandoItemId randoItemId = Rando::ConvertItem(randoSaveCheck.randoItemId);
+            Matrix_Scale(30.0f, 30.0f, 30.0f, MTXMODE_APPLY);
+            Rando::DrawItem(Rando::ConvertItem(randoSaveCheck.randoItemId, (RandoCheckId)CUSTOM_ITEM_PARAM), actor);
+        });
+}
+
+void Rando::ActorBehavior::InitEnemyDropBehavior() {
+    COND_VB_SHOULD(VB_ENEMY_DROP_COLLECTIBLE, IS_RANDO, {
+        Vec3f position = va_arg(args, Vec3f);
+
+        if (RANDO_SAVE_OPTIONS[RO_SHUFFLE_ENEMY_DROPS] == RO_GENERIC_OFF) {
+            return;
+        }
+
+        Actor* foundActor = FindActor(position, ACTORCAT_ENEMY);
+        if (foundActor == nullptr || foundActor->category != ACTORCAT_ENEMY) {
+            return;
+        }
+
+        int16_t actorId = foundActor->id;
+        switch (actorId) {
+            case ACTOR_EN_TALK_GIBUD:
+            case ACTOR_EN_RAILGIBUD:
+                actorId = ACTOR_EN_RD; // Gibdos
+                break;
+            case ACTOR_EN_RAIL_SKB:
+            case ACTOR_EN_HINT_SKB:
+                actorId = ACTOR_EN_SKB; // Stalchildren
+                break;
+            default:
+                break;
+        }
```

Comment:
After testing, I found that talking Gibdos and Oceanside/patrolling Stalchildren don't actually drop anything. In fact, well Gibdos don't die at all. They play dead for several seconds and then get back up. We can remove all of this.

```suggestion
        int16_t actorId = foundActor->id;
```